freqtrade/leverage/liquidation_price.py

```diff
@@ -41,147 +53,90 @@ def exception(
     exchange: str,
     trading_mode: TradingMode,
     collateral: Collateral,
-    margin_mode: Optional[MarginMode] = None
 ):
     """
         Raises an exception if exchange used doesn't support desired leverage mode
         :param exchange: Name of the exchange
-        :param margin_mode: one-way or hedge
         :param trading_mode: spot, margin, futures
         :param collateral: cross, isolated
     """
-    if not margin_mode:
-        raise OperationalException(
-            f"{exchange} does not support {collateral.value} {trading_mode.value} trading ")
 
     raise OperationalException(
-        f"{exchange} does not support {collateral.value} {margin_mode.value} Mode {trading_mode.value} trading ")
+        f"{exchange} does not support {collateral.value} Mode {trading_mode.value} trading ")
 
 
 def binance(
     open_rate: float,
     is_short: bool,
     leverage: float,
-    margin_mode: MarginMode,
     trading_mode: TradingMode,
     collateral: Collateral,
-    **kwargs
+    wallet_balance: float,
+    maintenance_margin_ex_1: float,
+    unrealized_pnl_ex_1: float,
+    maintenance_amount: float,
+    position_1: float,
+    entry_price_1: float,
+    maintenance_margin_rate: float,
 ):
     r"""
         Calculates the liquidation price on Binance
         :param open_rate: open_rate
         :param is_short: true or false
         :param leverage: leverage in float
-        :param margin_mode: one-way or hedge
         :param trading_mode: spot, margin, futures
         :param collateral: cross, isolated
 
-        :param \**kwargs:
-            See below
-
-        :Keyword Arguments:
-            * *wallet_balance* (``float``) --
-              Wallet Balance is crossWalletBalance in Cross-Margin Mode
-              Wallet Balance is isolatedWalletBalance in Isolated Margin Mode
-
-            * *maintenance_margin_ex_1* (``float``) --
-              Maintenance Margin of all other contracts, excluding Contract 1.
-              If it is an isolated margin mode, then TMM=0
-
-            * *unrealized_pnl_ex_1* (``float``) --
-              Unrealized PNL of all other contracts, excluding Contract 1.
-              If it is an isolated margin mode, then UPNL=0
-
-            * *maintenance_amount_both* (``float``) --
-              Maintenance Amount of BOTH position (one-way mode)
-
-            * *maintenance_amount_long* (``float``) --
-              Maintenance Amount of LONG position (hedge mode)
+        :param wallet_balance: Wallet Balance is crossWalletBalance in Cross-Margin Mode.
+          Wallet Balance is isolatedWalletBalance in Isolated Margin Mode
 
-            * *maintenance_amount_short* (``float``) --
-              Maintenance Amount of SHORT position (hedge mode)
+        :param maintenance_margin_ex_1: Maintenance Margin of all other contracts, excluding Contract 1.
```

Comment:
I think it might be better to shorten these variables to mm_ex_1, there just a lot of places where maintenance_margin is printed out and it clutters things up a bit. Keep the part to the right of the colon as Maintenance Margin though

Also, if this is our function to calculate maintenance margin, and it requires our other trades maintenance margins, then how does this function work. Is it some sort of induction type thing where we know the value is 0 when we enter our first trade, and then for our second trade we just take the value from the first one, and then we can keep doing that for each subsequent trade?

Reply:
Maintenance margin has its own slabs and formula to calculate I think we need a dict in another util file to generate this

Reply:
https://github.com/freqtrade/freqtrade/pull/5576/commits/da00b21cd50c1341dfd97b67e72257378d5ada74